webscrapbook/scrapbook/common/util.py

```diff
@@ -0,0 +1,121 @@
+import os, json, glob, re
+from math import ceil
+from collections import OrderedDict
+from itertools import islice
+
+# string functions
+###############################################################################
+
+def remove_prefix(text, prefix):
+    if text.startswith(prefix):
+        return text[len(prefix):]
+    return text
+
+def remove_suffix(text, suffix):
+    if text.endswith(suffix):
+        return text[:-1 * len(suffix)]
+    return text
+
+def remove_lines(file, count):
+    for _ in range(count):
+        file.readline()
+
+def parse_json(filepath, preprocessing):
+    data = {}
+    with open(filepath) as file:
+        json_string = preprocessing(file)
+        try:
+            data = json.loads(json_string, object_pairs_hook=OrderedDict)
+        except:
+            # TODO: better handling
+            raise Exception("Failed to parse json file " + filepath)
+    return data
+
+def get_number_suffix(text):
+    ''' return the number at the end of a string if no number found return empty string '''
+    return re.findall('([0-9]*$)', text)[0]
+
+###############################################################################
+
+
+# filesystem functions
+###############################################################################
+
+def add_directory_to_filename(directory, filename):
+    return os.path.join(directory, filename)
+
+def get_filename_no_ext(filepath):
+    return os.path.splitext(os.path.basename(filepath))[0]
+
+def find_file(directory, file_regex):
+    ''' given a directory, regex return a list of matching files 
+    
+        Parameters:
+            directory (str): directory containing files
+            file_regex (str): a regex to match a filename
+    '''
+    files = os.listdir(directory)
+    possible_files = [ add_directory_to_filename(directory, file) for file in files if re.match(file_regex, file)]
```

Comment:
We don't insert leading and trailing spacing between `(...)`, `[...]`, `{...}`, etc.

Reply:
I will remove all the unnecessary spacing.

Reply:
Fixed!